### PR TITLE
Using addQueryArgs to generate Manage All Reusable Blocks link

### PR DESCRIPTION
**Pull request by SofiaSousa**
## Description
Using `addQueryArgs` method  from `@wordpress/url` instead of hand coding 'Manage All Reusable Blocks' link.

## How has this been tested?
1. Open 'Reusable' panel in 'Add Block' menu
2. Check 'Manage All Reusable Blocks' link is redirecting to `edit.php?post_type=wp_block`.
2. Open 'More' menu (top right ...)
2. Check 'Manage All Reusable Blocks' link is redirecting to `edit.php?post_type=wp_block`.

## Types of changes
Reuse an existing method.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

File: packages/editor/src/components/inserter/menu.js [hunk 1]
```diff
@@ -290,7 +291,7 @@ export class InserterMenu extends Component {
 								className="editor-inserter__manage-reusable-blocks"
 								icon="admin-generic"
 								label={ __( 'Manage All Reusable Blocks' ) }
-								href="edit.php?post_type=wp_block"
+								href={ addQueryArgs( 'edit.php', { post_type: 'wp_block' } ) }
```
Comment (gziolo) [inline packages/editor/src/components/inserter/menu.js]: I don't think we should ever reference PHP file inside the `editor` package. I noticed there are other places where we link to WordPress specific url. @youknowriad - should we move out those bits to `edit-post` or make them customizable?

Comment (youknowriad) [reply packages/editor/src/components/inserter/menu.js]: There's an issue for that :) #9637

**Comment by youknowriad** [comment]
What's the benefit for using `addQueryArgs` if the URL is fixed?

File: packages/editor/src/components/inserter/menu.js (hunk 1 above)
Comment (gziolo) [reply packages/editor/src/components/inserter/menu.js]: Awesome :)

**Comment by SofiaSousa** [comment]
To be coherent. Then in our case, we override `addQueryArgs` in order to apply the changes we need on the editor paths since paths are not configurable.

**Comment by SofiaSousa** [comment]
@gziolo PR has the latest changes from the `master` now.

I did a little search for `getWPAdminURL` and figured out that is just an "[abstraction](https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/utils/url.js#L21)" of `addQueryArgs`. I don't know the reason for this, but probably there is a good one :)

Otherwise, I think we should replace all calls of `getWPAdminURL`.

**Comment by gziolo** [comment]
> I did a little search for getWPAdminURL and figured out that is just an "abstraction" of addQueryArgs. I don't know the reason for this, but probably there is a good one :)

I'm confused as much as you are, good finding :)

@youknowriad https://github.com/WordPress/gutenberg/commit/a55305602a8d99acb32834f3bb8958220e94c390
- can we deprecate `getWPAdminURL`? I would prefer if we had a way to abstract away all links and make them filterable if plugins want to change those links or being easy to override if projects like G in Drupal prefers to do so.

I think `addQueryArgs` is fine for now in that case.

File: package-lock.json [hunk 2]
```diff
@@ -2975,7 +2975,6 @@
 			"resolved": "https://registry.npmjs.org/align-text/-/align-text-0.1.4.tgz",
 			"integrity": "sha1-DNkKVhCT810KmSVsIrcGlDP60Rc=",
 			"dev": true,
-			"optional": true,
```
Comment (gziolo) [inline package-lock.json]: I see unrelated lock file changes in this PR. My bet is that it is because this file was regenerated using npm 6.7 which introduced some changes in a way `optional` flag is handled. Can you try to run `npm install` with the latest version of npm?

**Review by gziolo — CHANGES_REQUESTED** [review]
Other than mentioned lock file changes, these changes look good. Thanks for refreshing.

File: package-lock.json (hunk 2 above)
Comment (SofiaSousa) [reply package-lock.json]: Which version did you mean? I had npm@6.1.0 when those `"optional": true,` appeared. Tried to update to the latest (`npm install -g npm@latest`) but the latest version is npm@6.7.0.

Comment (gziolo) [reply package-lock.json]: > the latest version is npm@6.7.0

This one

Comment (SofiaSousa) [reply package-lock.json]: My bad, I misunderstood your comment.

**Review by gziolo — APPROVED** [review]
I opened #13653 to fix the issue with the lock file. I tested those changes and they work as expected. Thank you for opening this PR. I will land your changes from my branch, but technically it will still get merged under your name :)
